addons/survey/tests/test_certification_flow.py

```diff
@@ -150,6 +150,28 @@ def test_flow_certification(self):
         self.assertEqual(certification_email.attachment_ids[0].name, f'Certification - {certification.title}.html',
                          'Default certification report print_report_name is "Certification - %s" % (object.survey_id.display_name)')
 
+        # Check that the certification can be printed
+        with self.with_user('admin'):
+            new_company = self.env['res.company'].create({
+                'name': 'newB',
+            })
+            npu = self.env['res.users'].create({
+                'name': 'No print right user',
+                'login': 'npu',
+                'password': 'npu',
+                'groups_id': [(6, 0, [
+                        self.ref('base.group_user'),
+                        self.ref('survey.group_survey_user')
+                    ])],
+                'company_id': new_company.id,
+                'company_ids': [new_company.id],
+            })
+            for company in self.env['res.company'].search([]):
+                company.invalidate_recordset()
+            pdf = self.env['ir.actions.report'].with_user(npu).with_company(new_company)\
+                    ._render_qweb_pdf('survey.certification_report_view', res_ids=[user_inputs.id])[0]
+            self.assertTrue(pdf, 'Certification report should be generated')
```

Comment:
If it crashes, you never get to the last assertion, if it doesn't crash, it'll always succeed (or it's not related to what we want to test here).

Also, as you assume you can use `[user_inputs.id]` because it's one record, `user_inputs.ids` is shorter.  

```suggestion
            self.env['ir.actions.report'].with_user(npu).with_company(new_company)\
                    ._render_qweb_pdf('survey.certification_report_view', res_ids=user_inputs.ids)
```